modules/lotamePanoramaIdSystem.js

```diff
@@ -159,6 +163,169 @@ function getLotameLocalCache(clientId = undefined) {
   return cache;
 }
 
+function getConsentParameters(consentData) {
+  const params = {};
+  const usp = uspDataHandler.getConsentData();
+
+  let usPrivacy;
+  if (typeof usp !== 'undefined' && !isEmpty(usp) && !isEmptyStr(usp)) {
+    usPrivacy = usp;
+  }
+  if (!usPrivacy) {
+    // fallback to 1st party cookie
+    usPrivacy = getFromStorage('us_privacy');
+  }
+  if (usPrivacy) {
+    params.us_privacy = usPrivacy;
+  }  
+
+  let consentString;
+  if (consentData) {
+    if (isBoolean(consentData.gdprApplies)) {
+      params.gdpr_applies = consentData.gdprApplies;
+    }
+    consentString = consentData.consentString;
+  }
+  // If no consent string, try to read it from 1st party cookies
+  if (!consentString) {
+    consentString = getFromStorage('eupubconsent-v2');
+  }
+  if (!consentString) {
+    consentString = getFromStorage('euconsent-v2');
+  }
+  if (consentString) {
+    params.gdpr_consent = consentString;
+  }
+
+  return params;
+}
+
+function getCurrentPanoId(localCache) {
+  const hasExpiredPanoId = Date.now() > localCache.expiryTimestampMs;
+  if (!hasExpiredPanoId && !isEmpty(localCache.data)) {
+    return localCache.data;
+  }
+}
+
+function useCookieless() {
+  return navigator.userAgent && navigator.userAgent.indexOf('Safari') != -1 && navigator.userAgent.indexOf('Chrome') == -1;
+}
+
+function getRequestHost() {
+  return useCookieless() ? ID_HOST_COOKIELESS : ID_HOST;
+}
+
+function processResponseIDs({
+  profileId,
+  coreId,
+  clientId,
+  noConsent,
+  expiry,
+  consentIsErrorFree
+}) {
+  const hasCustomClientId = !isEmpty(clientId);
+  let storedCoreId;
+  if (hasCustomClientId) {
+    if (consentIsErrorFree) {
+      clearLotameCache(`${KEY_EXPIRY}_${clientId}`);
+    } else if (isStr(noConsent) && noConsent === 'CLIENT') {
+      saveLotameCache(`${KEY_EXPIRY}_${clientId}`, expiry, expiry);
+      // End Processing
+      return;
+    }
+  }
+
+  saveLotameCache(KEY_EXPIRY, expiry, expiry);
+
+  if (isStr(profileId)) {
+    if (consentIsErrorFree) {
+      setProfileId(profileId);
+    }
+
+    if (isStr(coreId)) {
+      saveLotameCache(KEY_ID, coreId, expiry);
+      storedCoreId = coreId;
+    } else {
+      clearLotameCache(KEY_ID);
+    }
+  } else {
+    if (consentIsErrorFree) {
+      clearLotameCache(KEY_PROFILE);
+    }
+    clearLotameCache(KEY_ID);
+  }
+  return storedCoreId;
+}
+
+function sendUserData(configParams, consentData) {
+  const clientId = configParams.clientId;
+  const hasCustomClientId = !isEmpty(clientId);
+  
+  let hem = isEmpty(configParams.hems) ? undefined : configParams.hems;
+  if (isArray(hem)) {
+    hem = hem.length > 0 ? hem[0] : undefined;
```

Comment:
this looks like you are sending one hem, but i thought you were intending to support any array?

Reply:
Correct. This is where the rubber meets the endpoint TBD road - I don't know what the request needs to look like on that side, so this is demonstrating passing only the first HEM. If the endpoint accepts multiples, then this can just as easily send them all. The important thing here is that the client-facing interface accepts an array.